fedbiomed/researcher/secagg/_secagg_context.py

```diff
@@ -539,6 +543,111 @@ def _create_payload_specific(self) -> Tuple[Union[dict, None], bool]:
         return context, True
 
 
+class SecaggKeyContext(SecaggContext):
+    """Handles a Additive Secret Sharing context on the Researcher side"""
+    def __init__(self, parties: List[str], experiment_id: str | None, secagg_id: str | None = None):
+        super().__init__(parties, experiment_id, secagg_id)
+
+        
+        self._element = SecaggElementTypes.SERVER_KEY
+        self._raise_if_missing_parties(parties, 2)
+        self._secagg_manager = _SKManager
+        #self._shares = []
+
+    def _matching_parties(self, context: dict) -> bool:
+        """Check if parties of given context are compatible with the secagg context element.
+
+        Args:
+            context: context to be compared with the secagg context element
+
+        Returns:
+            True if this context can be used with this element, False if not.
+        """
+        return matching_parties_servkey(context, self._parties)
+
+    def _create_payload_specific(self, context) -> Tuple[Union[dict, None], bool]:
+        """Researcher payload for creating Additive Secret Sharing secagg context element
+
+        Args:
+            context: specific secagg element context
+
+        Returns:
+            A tuple of a `context` and a `status` for the server key context element
+        """
+        # context = {'key': context,
+        #            'share': self._share.reconstruct()}
+
+        self._secagg_manager.add(self._secagg_id, self._parties, context, self._experiment_id)
+
+        logger.debug(
+            f"Additive Secret Sharing context successfully created for researcher_id='{environ['ID']}' "
+            f"secagg_id='{self._secagg_id}'")
+
+        return context, True
+
+    def setup(self) -> bool:
+        msg = ResearcherMessages.format_outgoing_message({
+            'researcher_id': self._researcher_id,
+            'secagg_id': self._secagg_id,
+            'element': self._element.value,
+            'experiment_id': self._experiment_id,
+            'parties': self._parties,
+            'command': 'secagg-additive-ss-setup-request',
```

Comment:
Hi @ybouilla,

I recommend you to use directly `AdditiveSSSetupRequest`, and since there is a defualt value for `command` you won't need to specify it.

Reply:
As you prefer, but the base class `setup` method doesnot provide `SecaggRequest`, but  uses the `ResearcherMessages.format_outgoing_message` instead. I can change both (`SecaggContext` and `SecaggKeyContext`)